returnn/datasets/distrib_files.py

```diff
@@ -355,39 +355,45 @@ def _distribute_evenly_by_size(
         assert len(files_per_bin) == num_bins
         bin_idx = 0
         size_taken = 0
+        total_size_taken = 0
         for i, f_tree in enumerate(files_order):
             size = file_sizes[_get_key_for_file_tree(f_tree)]
             num_remaining = len(files_order) - i
-            if num_remaining <= num_bins - bin_idx - 1:
-                # All remaining sub epochs must be filled.
-                assert files_per_bin[bin_idx]
+
+            try:
```

Comment:
Why this `try` block?

Reply:
Yeah, I'm not so sure about it. I'm using it to DRY the update to the average via the finally block. Otherwise with the `continue`s within the loop I'd have to repeat that code multiple times.

Reply:
No we should definitely not do this. It's like an obfuscated goto what you do here...